### PR TITLE
ChatSDK Performance: chatsdk.initialize()

### PR DESCRIPTION
**Pull request by vishnuvulichi**
This PR includes the following scenario.

chatsdk.initialize()

### CODE REVIEW

File: playwright/performance/chatsdk-initialize.spec.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                return request.url().includes(OmnichannelEndpoints.LiveChatConfigPath);
+            }),
+            page.waitForResponse(response => {
+                return response.url().includes(OmnichannelEndpoints.LiveChatConfigPath);
+            }),
+            await page.evaluate(async ({ omnichannelConfig }) => {
+                const { OmnichannelChatSDK_1: OmnichannelChatSDK } = window;
+                const chatSDK = new OmnichannelChatSDK.default(omnichannelConfig);
+
+                let starttime = new Date();
```
Comment (xTEddie) [inline playwright/performance/chatsdk-initialize.spec.ts:24]: `startTime`

File: playwright/performance/chatsdk-initialize.spec.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                await chatSDK.initialize();
+                let endtime = new Date();
```
Comment (xTEddie) [inline playwright/performance/chatsdk-initialize.spec.ts:26]: `endTime`

File: playwright/performance/chatsdk-initialize.spec.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                let timeTaken = endtime.getTime() - starttime.getTime();
+
+                const runtimeContext = {
+                    requestId: chatSDK.requestId
```
Comment (xTEddie) [inline playwright/performance/chatsdk-initialize.spec.ts:30]: Please add `timeTaken` as part of `runtimeContext` instead

File: playwright/performance/chatsdk-initialize.spec.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import OmnichannelEndpoints from '../utils/OmnichannelEndpoints';
+
+const testPage = fetchTestPageUrl();
+const omnichannelConfig = fetchOmnichannelConfig('UnauthenticatedChat');
+
+test.describe('ChatSDK Performance: ', () => {
+    test.only('ChatSDK.initialize()', async ({ page }) => {
+        await page.goto(testPage);
+        
+        let [request, response, [runtimeContext,timeTaken] ] = await Promise.all([
```
Comment (xTEddie) [inline playwright/performance/chatsdk-initialize.spec.ts:13]: Please flatten out `timeTaken` by add it as part of `runtimeContext`

Comment (xTEddie) [reply playwright/performance/chatsdk-initialize.spec.ts:13]: `request` is not needed

File: playwright/performance/chatsdk-initialize.spec.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                    requestId: chatSDK.requestId
+                };
+                
+                return [runtimeContext, timeTaken];
+            }, { omnichannelConfig }),
+        ]);
+
+        console.log("chatSDK.initialize(): " + timeTaken);
+        
+        const { requestId } = runtimeContext;
```
Comment (xTEddie) [inline playwright/performance/chatsdk-initialize.spec.ts:39]: You should be able to retrieve `timeTaken` from `runtimeContext` after flattening it out

File: playwright/performance/chatsdk-initialize.spec.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        const requestUrl = `${omnichannelConfig.orgUrl}/${OmnichannelEndpoints.LiveChatConfigPath}/${omnichannelConfig.orgId}/${omnichannelConfig.widgetId}?requestId=${requestId}&channelId=lcw`;
```
Comment (xTEddie) [inline playwright/performance/chatsdk-initialize.spec.ts:40]: We won't need this after removing `waitForRequest`

File: playwright/performance/chatsdk-initialize.spec.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+        expect(request.url() === requestUrl).toBe(true);
```
Comment (xTEddie) [inline playwright/performance/chatsdk-initialize.spec.ts:42]: We could remove this after removing `waitForRequest`

File: playwright/performance/chatsdk-initialize.spec.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
+import fetchOmnichannelConfig from '../utils/fetchOmnichannelConfig';
+import fetchTestPageUrl from '../utils/fetchTestPageUrl';
+import { test, expect } from '@playwright/test';
+import OmnichannelEndpoints from '../utils/OmnichannelEndpoints';
+
+const testPage = fetchTestPageUrl();
+const omnichannelConfig = fetchOmnichannelConfig('UnauthenticatedChat');
+
+test.describe('ChatSDK Performance: ', () => {
```
Comment (xTEddie) [inline playwright/performance/chatsdk-initialize.spec.ts:9]: `ChatSDK.initialize()` should be good enough

File: playwright/performance/chatsdk-initialize.spec.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    test.only('ChatSDK.initialize()', async ({ page }) => {
```
Comment (xTEddie) [inline playwright/performance/chatsdk-initialize.spec.ts:10]: `test.only()` can be removed

File: playwright/performance/Initialize.spec.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import fetchOmnichannelConfig from '../utils/fetchOmnichannelConfig';
+import fetchTestPageUrl from '../utils/fetchTestPageUrl';
+import { test, expect } from '@playwright/test';
+import OmnichannelEndpoints from '../utils/OmnichannelEndpoints';
+
+const testPage = fetchTestPageUrl();
+const omnichannelConfig = fetchOmnichannelConfig('UnauthenticatedChat');
+
+test.describe('ChatSDK.initialize(): ', () => {
```
Comment (xTEddie) [inline playwright/performance/Initialize.spec.ts:9]: `Performance @Performance`

Comment (vishnuvulichi) [reply playwright/performance/Initialize.spec.ts]: changed

**Comment by vishnuvulichi** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by vishnuvulichi** [comment]
> @vishnuvulichi please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree company="Microsoft"

File: playwright/performance/Initialize.spec.ts [hunk 11]
```diff
@@ -0,0 +1,38 @@
+import fetchOmnichannelConfig from '../utils/fetchOmnichannelConfig';
+import fetchTestPageUrl from '../utils/fetchTestPageUrl';
+import { test, expect } from '@playwright/test';
+import OmnichannelEndpoints from '../utils/OmnichannelEndpoints';
+
+const testPage = fetchTestPageUrl();
+const omnichannelConfig = fetchOmnichannelConfig('UnauthenticatedChat');
+
+test.describe('Performance @Performance: ', () => {
```
Comment (xTEddie) [inline playwright/performance/Initialize.spec.ts:9]: Please remove `:` from `@Performance:`. `@Performance` would be a tag to be used to run these tests.

Comment (vishnuvulichi) [reply playwright/performance/Initialize.spec.ts]: removed

**Review by xTEddie — APPROVED** [review]

**Pull request merged**